test/Test/Iris/Interactive/Question.hs

```diff
@@ -0,0 +1,29 @@
+module Test.Iris.Interactive.Question (questionSpec) where
+
+import Data.Text (Text)
+import Test.Hspec (Spec, SpecWith, describe, it, shouldMatchList)
+
+import Iris.Interactive.Question (
+    -- under test
+    YesNo (..),
+    parseYesNo,
+ )
+
+yesAnswers :: [Text]
+yesAnswers = "y" : "Y" : [y <> e <> s | y <- ["y", "Y"], e <- ["e", "E", ""], s <- ["s", "S"]]
+
+questionSpec :: Spec
+questionSpec =
+    describe "Question - parse YesNo" $ do
+        checkElements yesAnswers (Just Yes)
+        checkElements ["n", "N", "NO", "no", "No", "nO"] (Just No)
+        checkElements ["", "a", "ye", "NOone", "yesterday", "oui"] Nothing
+
+checkElements
+    :: [Text]
+    -> Maybe YesNo
+    -> SpecWith ()
+checkElements values expected = do
+    describe ("elementes should parse to " ++ show expected) $ do
+        it (show values) $ do
+            map (\x -> (x, parseYesNo x)) values `shouldMatchList` map (,expected) values
```

Comment:
I actually liked the previous implementation where each constructor was a separate test-case. It's nice that the testing framework shows proper diffs of lists.

But I find the output with string quotes in lists too noisy and having a single test case per string is actually fine in this particular case because there's a limited number of test cases we want to parse.

So I would rewrite this code like this:
```suggestion
    describe "Question - parse YesNo" $ do
        checkElements yesAnswers (Just Yes)
        checkElements ["n", "N", "NO", "no", "No", "nO"] (Just No)
        checkElements ["", "a", "ye", "NOone", "yesterday", "oui"] Nothing

checkElements
    :: [Text]
    -> Maybe YesNo
    -> SpecWith ()
checkElements values expected = do
    describe ("should parse to " ++ show expected) $
        forM_ values $ \strValue ->
            it (Text.unpack strValue) $
                parseYesNo strValue `shouldBe` expected
```